res/skins/Shade/skin.xml

```diff
@@ -427,14 +454,36 @@
           <SetVariable name="samplernum3">5</SetVariable>
           <SetVariable name="samplernum4">6</SetVariable>
         </Template>
-
         <Template src="skin:samplerrow.xml">
           <SetVariable name="row">2</SetVariable>
           <SetVariable name="samplernum1">3</SetVariable>
           <SetVariable name="samplernum2">4</SetVariable>
           <SetVariable name="samplernum3">7</SetVariable>
           <SetVariable name="samplernum4">8</SetVariable>
         </Template>
+
+        <!-- minimal decks, visible with maximized library -->
+        <Template src="skin:decks_row_small.xml"/>
+
+        <!-- put invisble waveforms Singletons here to enable end-of-track warning
+            without actually showing scrolling waveforms -->
+        <WidgetGroup>
+          <Layout>horizontal</Layout>
+          <Size>0me,0f</Size>
```

Comment:
This does not work. The waveform width is reported as 5xx and visible. The width changes from 3xx when switching library though.
So it is tendered still rendered.

Do you have an idea why this is required for the end of Treck Warnung? We need probably an alternative solution.

Reply:
yeah, as stated above this is just a hack (1) for the end-of-track warning.
and I recall a bug (2) where needed these invisible waveforms to keep the rest of the GUI responsove with Intel graphics when waveforms are hidden in Tango (which we now apply to all skins to have only minimal decks & library).
https://bugs.launchpad.net/mixxx/+bug/1758660
https://bugs.launchpad.net/mixxx/+bug/1204294
https://bugs.launchpad.net/mixxx/+bug/1801636

If someone could look into the issue, that would be very nice as we could remove some skin hacks if it's fixed.

Reply:
>  The waveform width is reported as 5xx and visible.

can you actually see it in Shade? or is debug telling the widget is visible?